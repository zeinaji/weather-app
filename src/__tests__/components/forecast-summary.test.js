import React from "react";
import { cleanup, render } from "@testing-library/react";
import ForecastSummary from "../../components/forecast-summary";

afterEach(cleanup);

describe("ForecastSummary", () => {
  it("renders correctly", () => {
    const { asFragment } = render(
      <ForecastSummary
        date="mockDate"
        temperature="mockTemperature"
        description="mockDescription"
        icon="mockIcon"
      />
    );

    expect(asFragment).toMatchSnapshot();
  });

  it("renders the correct props", () => {
    const { getByText } = render(
      <ForecastSummary
        date="mockDate"
        temperature="mockTemperature"
        description="mockDescription"
        icon="mockIcon"
      />
    );

    const date = getByText("mockDate");
    expect(date.closest("div")).toHaveClass("date");

    const temperature = getByText("mockTemperatureÂ°C");
    expect(temperature.closest("div")).toHaveClass("temperature");

    const description = getByText("mockDescription");
    expect(description.closest("div")).toHaveClass("description");

    const icon = getByText("mockIcon");
    expect(icon.closest("div")).toHaveClass("icon");
  });
});
