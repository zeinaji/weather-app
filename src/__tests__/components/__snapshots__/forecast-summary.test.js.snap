// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForecastSummary renders correctly 1`] = `[Function]`;
